features/facebookImport/src/model/analyses/activities-analysis.js

```diff
@@ -0,0 +1,76 @@
+import React from "react";
+import i18n from "../../i18n.js";
+import RootAnalysis from "./root-analysis.js";
+
+import ActivitiesMinistory from "../../components/activitiesMinistory/activitiesMinistory.jsx";
+
+export default class ActivitiesAnalysis extends RootAnalysis {
+    get title() {
+        return i18n.t("activitiesMinistory:title");
+    }
+
+    async analyze({ facebookAccount }) {
+        const activities = {
+            timestamp: [
+                ...facebookAccount.followedPages,
+                ...facebookAccount.friends,
+                ...facebookAccount.interactedAdvertisers,
+                ...facebookAccount.likedPages,
+                ...facebookAccount.receivedFriendRequests,
+                ...facebookAccount.recommendedPages,
+                ...facebookAccount.searches,
+                ...facebookAccount.unfollowedPages,
+            ],
+            timestamp_ms: [],
+        };
+
+        //for nested structures
+        facebookAccount.offFacebookCompanies.forEach((company) =>
+            activities.timestamp.push(...company.events)
+        );
+
+        facebookAccount.messageThreads.forEach((thread) =>
+            activities.timestamp_ms.push(...thread.messages)
+        );
+
+        let total = { total: 0 };
+        for (let [timestampKey, activitiesValues] of Object.entries(
+            activities
+        )) {
+            for (let activity of activitiesValues) {
+                const timeOfActivity = new Date(activity[timestampKey]);
+                const activityYear = timeOfActivity.getFullYear();
+                if (activityYear >= 2006) {
```

Comment:
Is the check `activityYear >= 2006` because of inconsistencies in the data?

Reply:
Exactly, when processing I got some 1970s dates, which would mean the timestamps are too small in the order of magnitude I think

Reply:
Do you have some examples of how those values look like inside the export?

Reply:
For Example timestamp "1569531600", which gets computed to 1970 by Date(), yet here for example its 2019 https://timestamp.online/. Other timestamps look like this "1417123972786", which is 3 digits longer and works with Date() but not with the online tool (year 46876 😅 )

Reply:
Ok I think one is the 10 digit and the other the 13 digit unix time epoch timestamp. Will find a way to parse them differently.